src/cache_inode/cache_inode_misc.c

```diff
@@ -822,6 +834,14 @@ cache_inode_release_dirents(cache_entry_t *entry,
 			entry->object.dir.nbactive = 0;
 			atomic_clear_uint32_t_bits(&entry->flags,
 						   CACHE_INODE_DIR_POPULATED);
+			if (entry->obj_handle->attrs->filesize >
```

Comment:
You should blindly delete all entries. Do not look at the cache flag.

Reply:
atta_lock and content_lock are protecting the file size from being modified while we do readdir.